pyomo/repn/plugins/gams_writer.py

```diff
@@ -252,13 +252,21 @@ def _write_model(self,
         for con in model.component_data_objects(Constraint,
                                                 active=True,
                                                 sort=sort):
-            if skip_trivial_constraints and con.body.is_fixed():
+
+            if (not con.has_lb()) and \
+               (not con.has_ub()):
+                assert not con.equality
+                continue # non-binding, so skip
```

Comment:
```python
>>> import pyomo.kernel as pmo
>>> c = pmo.constraint(body=0)
>>> c.has_lb()
False
>>> c.has_ub()
False
```
I believe you can also arrive at this with the AML interface by just setting the lower or upper attributes to None (or +/-inf) after the constraint is built.